api/sip/client.py

```diff
@@ -351,7 +352,9 @@ def login(self):
                 self.location_code,
             )
             if response["login_ok"] != "1":
-                raise OSError("Error logging in: %r" % response)
+                raise PatronAuthorizationFailedException(
+                    "Error logging in: %r" % response
+                )
```

Comment:
In case the credentials provided are incorrect, it should not contribute to the failure counts of the auth mechanism, hence this exception type was changed to differentiate the login failure vs API failure

Reply:
A failed login would usually mean the API server is running just fine and the User has incorrectly input their credentials. This should not impact other Users on the system who would be using the functioning API for authentication.

The response being logged and returned in the Exception is a json value, we cannot reliably depend on it string format to differentiate the types. I also believe having a different Exception type altogether is more explicit and future-proof as opposed to differentiating based on the string messages within them.

Reply:
I was focused in on SIP2 service, where login is distinct from patron lookup. In that (SIP2) case, login failure would, in a practical sense, be tantamount to service unavailability. But that is not the case for all of our integrations.